feature/src/main/java/com/teamdontbe/feature/signup/SignUpProfileActivity.kt

```diff
@@ -26,68 +29,109 @@ class SignUpProfileActivity :
+        window.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN)
+    }
+
+    // 닉네임 중복 체크 메세지
+    private fun initUpdateErrorMessage(flag: String) {
         viewModel.isNickNameValid.observe(this) {
             val messageResId =
                 if (it) R.string.sign_up_profile_check_text else R.string.sign_up_profile_correct_check
             val textColorResId = if (it) R.color.gray_8 else R.color.error
 
-            updateAgreeMessage(messageResId, textColorResId)
+            updateAgreeText(messageResId, textColorResId)
+            initDoubleBtnClickListener(flag, it)
+        }
+    }
+
+    private fun updateAgreeText(
+        messageResId: Int,
+        textColorResId: Int,
+    ) {
+        binding.tvSignUpAgreeMessage.apply {
+            text = context.getString(messageResId)
+            setTextColor(colorOf(textColorResId))
         }
     }
 
-    private fun initDoubleBtnClickListener(flag: Int) {
+    // 중복 확인 버튼 클릭
+    private fun initDoubleBtnClickListener(flag: String, btnAvailable: Boolean) {
         binding.btnSignUpProfileDoubleCheck.setOnClickListener {
-            val inputNickName = binding.etSignUpProfileNickname.text.toString()
-            viewModel.getNickNameDoubleCheck(inputNickName)
-            nextBtnObserve(inputNickName, flag)
+            viewModel.getNickNameDoubleCheck(binding.etSignUpProfileNickname.text.toString())
+            handleFlag(flag, btnAvailable)
         }
     }
 
-    private fun updateErrorMessage(doubleCheck: Boolean) {
-        val messageResId =
-            if (doubleCheck) R.string.sign_up_profile_use_posssible else R.string.sign_up_profile_use_impossible
-        val textColorResId = if (doubleCheck) R.color.primary else R.color.error
+    // 프로필 편집, 마이페이지 프로필 편집 분기처리
+    private fun handleFlag(flag: String, btnAvailable: Boolean) {
+        if (btnAvailable) {
+            when (flag) {
+                SIGN_UP_AGREE -> hideKeyboard(binding.root)
+                MY_PAGE_PROFILE -> focusOnIntroduceEditText()
+            }
+        }
+    }
 
-        updateAgreeMessage(messageResId, textColorResId)
+    // editText 자동 포커스
+    private fun focusOnIntroduceEditText() {
+        binding.etSignUpAgreeIntroduce.apply {
+            requestFocus()
+            setSelection(this.text.length)
+        }
     }
 
-    private fun initMyPageStateObserve() {
+    private fun initNickNameDoubleStateObserve() {
         viewModel.nickNameDoubleState.flowWithLifecycle(lifecycle).onEach {
             when (it) {
                 is UiState.Loading -> Unit
                 is UiState.Success -> {
-                    if (it.data.isEmpty()) {
-                        updateErrorMessage(false)
-                    } else {
+                    if (it.data) {
                         updateErrorMessage(true)
+                    } else {
+                        updateErrorMessage(false)
                     }
```

Comment:
```suggestion
                    updateErrorMessage(it.data)
```
P2 : 요렇게 하면 더 간결해질 수 있겠네요!

Reply:
이거 좋은ㄴ생각입니다!
[a0e064c](https://github.com/TeamDon-tBe/ANDROID/pull/189/commits/a0e064c0b74ef6359807fa918a50956426dc9ce6)